pypxe/tftp.py

```diff
@@ -123,13 +129,16 @@ def read(self, address, message):
             self.logger.error("Mode '{mode}' not supported".format(mode = mode))
             self.tftpError(address, 5, 'Mode {mode} not supported'.format(mode = mode))
             return
-        filename = self.filename(message)
+        req_file = self.filename(message)
+        # avoid directory traversal: strip all ../ and make it relative
+        filename = os.path.normpath(os.sep + os.getcwd() + os.sep + req_file).lstrip(os.sep)
```

Comment:
Just a note, it is possible to refer to paths in python on Windows in the same way as Unix (`os.listdir('/')` works as expected).

A warning would be good, although in `logger` terms it should probably be `critical`.

I'm currently investigating other Windows related issues and will have an Issue posted shortly.